packages/cli/src/util/logs.ts

```diff
@@ -30,11 +38,217 @@ export function displayBuildLogs(
       quiet: false,
       findOpts: { direction: 'forward', follow },
     },
     abortController
   );
   return { promise, abortController };
 }
+
+export interface DisplayRuntimeLogsOptions {
+  projectId?: string;
+  deploymentId: string;
+  parse?: boolean;
+}
+
+const runtimeLogSpinnerMessage = `waiting for new logs...`;
+
+export interface RuntimeLog {
+  level: 'error' | 'warning' | 'info';
+  message: string;
+  rowId: string;
+  source:
+    | 'delimiter'
+    | 'edge-function'
+    | 'edge-middleware'
+    | 'serverless'
+    | 'request';
+  timestampInMs: number;
+  domain: string;
+  messageTruncated: boolean;
+  requestMethod: string;
+  requestPath: string;
+  responseStatusCode: number;
+}
+
+export async function displayRuntimeLogs(
+  client: Client,
+  options: DisplayRuntimeLogsOptions,
+  abortController: AbortController
+): Promise<number> {
+  const { log, debug, print, spinner, stopSpinner, warn } = client.output;
+  const { projectId, deploymentId, parse } = options;
+
+  const query = new URLSearchParams({ format: 'lines' });
+
+  const url = `/v1/projects/${projectId}/deployments/${deploymentId}/runtime-logs?${query}`;
+  spinner(runtimeLogSpinnerMessage);
+  const timeout = setTimeout(() => {
+    abortController.abort();
+    warn(
+      `${chalk.bold(
+        `Command automatically interrupted after ${CommandTimeout}.`
+      )}\n`
+    );
+  }, ms(CommandTimeout));
+
+  const response = await client.fetch(url, {
+    json: false,
+    // @ts-expect-error: typescipt is getting confused with the signal types from node (web & server) and node-fetch (server only)
+    signal: abortController.signal,
+    retry: {
+      retries: 3,
+      onRetry: err => {
+        log(`Runtime logs error: ${err.message}`);
+        if (err instanceof Error && err.name === 'AbortError') {
+          return;
+        }
+        throw err;
+      },
+    },
+  });
+  // handle the event stream and make the promise get rejected
+  // if errors occur so we can retry
+  return new Promise<number>((resolve, reject) => {
+    const stream = response.body.pipe(parse ? jsonlines.parse() : split());
+    let finished = false;
+    let errored = false;
+
+    function finish(err?: unknown) {
+      if (finished) return;
+      clearTimeout(timeout);
+      stopSpinner();
+      finished = true;
+      if (err) {
+        reject(err);
+      } else {
+        resolve(abortController.signal.aborted ? 1 : 0);
+      }
+    }
+
+    const handleData = (data: RuntimeLog | string) => {
+      let log: RuntimeLog = parse ? data : JSON.parse(data as string);
+      if (isRuntimeLimitDelimiter(log)) {
+        abortController.abort();
+        warn(`${chalk.bold(log.message)}\n`);
+        return;
+      }
+      // eslint-disable-next-line no-console -- we intent to write unparsed logs to stdout so JQ could read them
+      parse ? prettyPrintLogline(log, print) : console.log(data);
```

Comment:
The intent of that eslint rule is so that `client.output.log` is used here instead of `console.log`

Reply:
I wish we could use `client.output.log`, but this will print to the wrong stream. We need the data to be printed on stdout (so it could be piped into `jq` and other unix tools), but `client.output [is configured to print on stderr](https://github.com/vercel/vercel/blob/damien/log-457-runtime-logs-on-vc/packages/cli/src/index.ts#L110), regardless of the level used.

Reply:
Ah ok I see, we also have `client.stoud`, we mainly want to keep this scoped to the client so we can make changes in the future more easily.

Reply:
Done!! Thanks for the tip